### PR TITLE
Migrate to bzlmod

### PR DESCRIPTION
**Pull request by keith**
This switches over to using bzlmod for local testing, as well as creates a MODULE.bazel file that we can push to the central registry so other projects can depend on yams directly.

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module(
+    name = "yams",
```
Comment (keith) [inline MODULE.bazel:2]: unfortunately bzlmod is prescriptive about this and you cannot have capital letters

**Comment by jpsim** [comment]
> creates a MODULE.bazel file that we can push to the central registry

Do I need to do something to do this?

File: MODULE.bazel (hunk 1 above)
Comment (jpsim) [reply MODULE.bazel:2]: it's fine

**Comment by jpsim** [comment]
Thanks! Certainly makes this case simpler.

**Comment by keith** [comment]
idk how much you want to sign up for for that, im fine to do submit it if you'd like, there is also automation we can configure with a github app for auto submitting updates

**Pull request merged**